### DISCUSSION TITLE
Custom theme without less

### DISCUSSION THREAD

**Discussion by sdt-jchau** [ant-design/ant-design#42562] [General]
We highly customized the css styles in v4.  When upgrading to v5, we want to follow Ant standards and use the new JS theme tokens.  However, is there recommendation for overriding styles that are not available via token?  For example, we want to change the default button border color from gray to light blue.  Is that possible without css override??

**Comment by bombillazo** [comment]
Not all styles are available via tokens. If the API does not expose them, customizing the theme via CSS overrides is the only way.

**Reply by sdt-jchau** [reply]
Except customizing via CSS is harder now because Ant v5 uses CSS-in-JS.  I like where Ant is going with design tokens but a lot of components are missing tokens that were previously available as less variables.

**Comment by BoyYangzai** [comment]
@sdt-jchau 
Hi bro! 
We are working hard on the migration and believe we will meet soon !
current progression：https://github.com/ant-design/ant-design/issues/41884
And you are welcome to join the migration contribution😄
